### PR TITLE
Timestamp Mods

### PR DESCRIPTION
**Pull request by herostrat**
A work in progress for advanced options for timestamp modification.

# Workpackages:

 - [x] Add drawing helper 
   - [x] Color
   - [x] Scaling
   - [x] Effect (shadow & solid background
   - [x] Thickness
   - [x] Date format
 - [x] Change all instances of timestamp decoration with helper
 - [x] Add option in config
 - [x] Documentation

# Images:

### Ugly stuff for your plessure
mixup1
### Shadow, B/W and top left
shadow
### Solid, B/W and bot left
solid_bl
### Solid, B/W and bot right
solid_br
### Solid, B/W and top left
solid_tl
### Solid, B/W and top right
solid_tr
### Scaling and ugly colors
grafik

Closes blakeblackshear/frigate/issues/968
Closes https://github.com/blakeblackshear/frigate/issues/284
Closes https://github.com/blakeblackshear/frigate/issues/400

### CODE REVIEW

**Comment by herostrat** [comment]
For the next working packages I will wait until/if https://github.com/blakeblackshear/frigate/pull/854 gets merged, as it affects the configuration.

**Comment by retrospectsecurity** [comment]
Can you add 'OSD' as a programmable option via webhook or something? Would be great to integrate sensor data on the overlay.

**Comment by Minglarn** [comment]
That would be awesome.

**Comment by herostrat** [comment]
Can you give me a link what OSD means in this case? I am not familiar with it.

**Comment by Minglarn** [comment]
OSD: On Screen Display. 
Let's say that you wanted moisture or actual temperature in your clips?

**Comment by blakeblackshear** [comment]
This timestamp will not make it into clips. Only snapshots and the live mjpeg view.

**Comment by blakeblackshear** [comment]
@herostrat I don't expect #854 to be merged anytime soon, so you may want to go ahead and incorporate this into the the current config options.

**Comment by herostrat** [comment]
I think this is ready for review/test.

File: frigate/object_processing.py [hunk 1]
```diff
@@ -239,25 +244,17 @@ def get_jpg_bytes(
...
-                self.thumbnail_data["frame_time"]
-            ).strftime("%m/%d/%Y %H:%M:%S")
-            size = cv2.getTextSize(
-                time_to_show, cv2.FONT_HERSHEY_SIMPLEX, fontScale=1, thickness=2
-            )
-            text_width = size[0][0]
-            desired_size = max(150, 0.33 * best_frame.shape[1])
-            font_scale = desired_size / text_width
-            cv2.putText(
+            logger.info("Jetzad")
```
Comment (blakeblackshear) [inline frigate/object_processing.py:248]: Should this be removed?

Comment (herostrat) [reply frigate/object_processing.py]: Yes, this was an artifact from debugging. Sorry

**Comment by herostrat** [comment]
It is in the branch release-0.9.0 as all new developed features atm.

**Comment by JasonSwindle** [comment]
Did this release? It is not in the docs, and 0.9.0 is public.

**Pull request merged**